### DISCUSSION TITLE
Managing KubeVela Addons with Duplicate Names

### DISCUSSION THREAD

**Discussion by Jayashree138** [kubevela/kubevela#6285] [Q&A]
Hey,
While working on a KubeVela addon, I've noticed a curious behavior where multiple addons with the same name are generated with respect to the number of files in `resource` folder containing Application YAML files. When I execute the `vela addon disable` command, all these duplicate copies are removed. I'm curious to understand why this happens and what it means for managing KubeVela addons.

**Comment by Jayashree138** [comment]
example: 
![image](https://github.com/kubevela/kubevela/assets/67368678/dee71f8f-70e1-4ad7-8db1-b03ca7900588)
resource file :
![image](https://github.com/kubevela/kubevela/assets/67368678/131af0c8-b9e2-4d34-8ed3-6e6ee258305f)

**Comment by chivalryq** [comment]
Can you show the content of application YAML?

**Comment by chivalryq** [comment]
And what's the use case for putting application in resources?

**Comment by Jayashree138** [comment]
Use Case: We have 2 addons where in first addon creates the definitions and the second addon is used to deploy the application files. The application files are kept in the resource folder as it only applies once when the addon is created. The two addons have dependencies specified in order to have the right workflow.

**Reply by chivalryq** [reply]
I see, but can you check the application's name in resources?

**Reply by Jayashree138** [reply]
The application names are different:

application example:
```
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: application1
spec:
  components:
    - name: app1
    -type: cluster-resource-1 # CRD from addon 1
```
    One more clarity: 
    Addon 1: Creates CRDs (definition)
    Addon 2: Creates application with type created by addon1.

**Reply by chivalryq** [reply]
If the application name differs and after enabling addon there are three addon listed with the same name (`test-addon`), there might be a bug in `vela addon list` or `vela addon enable`. You can check if the application name in cluster is as expected. There should be two apps: `addon-test` and `application1`
